plat/allwinner/common/sunxi_common.c

```diff
@@ -71,3 +74,25 @@ uint16_t sunxi_read_soc_id(void)
 
 	return reg >> 16;
 }
+
+const char *sunxi_get_dt_name(void)
+{
+	const struct spl_head *spl_head =
+		(const struct spl_head *) SUNXI_SRAM_A1_BASE;
```

Comment:
U-Boot SPL already has a mechanism (CONFIG_SPL_ATF) for passing the device tree and other data to ATF. We should use that instead of hardcoding addresses and checking for magic values.

Reply:
But the SPL header method is already well defined, and being used by SPL and U-Boot.

It's not a magic.

Reply:
I agree with @smaeul that we should do it properly now. While the SPL header method might be somewhat "well defined", checking against all known DT names is not going to fly. I just added this hack to the older ATF port because it provided a pragmatic solution, and the code base was hacked anyway.

So what about the following:
1) We drop the whole SPL header DT string code from this PR.
2) We assume an AXP805 if an H6 is detected, keeping the warning or maybe tone it down to just an announcement that we will use an AXP805.
2a) We could possibly try to detect the AXP805, either using I2C methods to check for a device on the bus or by querying and verifying the ID from the AXP805.
3) Later we add DT support to this ATF port, piggy backing on the Rockchip work, probably.
4) We then scan the DT for the AXP805 compatible string or something equally easy, and use this information to enable the PMIC code.

So 1) and 2) means actually removing code from this PR, and we address 3) and 4) separately later.

Does that make sense?

Reply:
I choose to follow 2a, although 2 will be used temporarily.

I have already seen H6+AXP802 sample board photo (although no real board).

BTW AXP802 is customized AXP803.